components/experimental/src/duration/format.rs

```diff
@@ -0,0 +1,528 @@
+                // i. Assert: The Unit value of the current row is "milliseconds", "microseconds", or "nanoseconds".
+                // ii. Let value be the value of duration's field whose name is the Value Field value of the current row.
+                // iii. Set value to value / (10 ^ exponent).
+                // iv. Set result to result + value.
+                result *= u128::pow(10, 3);
+                result += u128::from(value);
+                // v. Set exponent to exponent + 3.
+                exponent += 3;
+            }
+        }
+
+        FixedDecimal::from(result).multiplied_pow10(-exponent)
+    }
+
+    /// Section 1.1.11
+    /// Formats numeric seconds to sink. Requires seconds formatting style to be either Numeric or TwoDigit.
+    fn format_numeric_seconds<V: PartsWrite + ?Sized>(
+        &self,
+        mut second_val: FixedDecimal,
+        sign_displayed: &mut bool,
+        minute_formatted: bool,
+        sink: &mut V,
+    ) -> core::fmt::Result {
+        // 1. Let secondsStyle be durationFormat.[[SecondsStyle]].
+        // 2. Assert: secondsStyle is "numeric" or secondsStyle is "2-digit".
+        debug_assert!(matches!(
+            self.fmt.options.second,
+            FieldStyle::Numeric | FieldStyle::TwoDigit
+        ));
+
+        // 3. Let result be a new empty List.
+
+        // 4. If minutesDisplayed is true, then
+        if minute_formatted {
+            // a. Let separator be durationFormat.[[DigitalFormat]].[[MinutesSecondsSeparator]].
+            // b. Append the Record { [[Type]]: "literal", [[Value]]: separator, [[Unit]]: empty } to result.
+            sink.with_part(parts::LITERAL, |e| {
+                e.write_str(self.fmt.digital.get().separator.as_ref())
+            })?;
+        }
+
+        // 5. Let nfOpts be OrdinaryObjectCreate(null).
+        // 6. Let numberingSystem be durationFormat.[[NumberingSystem]].
+        // 7. Perform ! CreateDataPropertyOrThrow(nfOpts, "numberingSystem", numberingSystem).
+
+        // 8. If secondsStyle is "2-digit", then
+        if self.fmt.options.second == FieldStyle::TwoDigit {
+            // a. Perform ! CreateDataPropertyOrThrow(nfOpts, "minimumIntegerDigits", 2𝔽).
+            second_val.pad_start(2);
+        }
+
+        // 9. If signDisplayed is false, then
+        if !*sign_displayed {
+            // a. Perform ! CreateDataPropertyOrThrow(nfOpts, "signDisplay", "never").
+            second_val.apply_sign_display(SignDisplay::Never);
+        }
+
+        // 10. Let nf be ! Construct(%NumberFormat%, « durationFormat.[[Locale]], nfOpts »).
+
+        match self.fmt.options.fractional_digits {
+            // 11. If durationFormat.[[FractionalDigits]] is undefined, then
+            FractionalDigits::ShowAll => {
+                // a. Let maximumFractionDigits be 9𝔽.
+                // b. Let minimumFractionDigits be +0𝔽.
+                second_val.trunc(-9);
+                second_val.pad_end(-1);
```

Comment:
I think you don't need this line, but if you have it, it should be just

```suggestion
                second_val.pad_end(0);
```

Reply:
`pad_end` has no effect on non-negative position arguments.

`pad_end` here removes possible trailing zeroes that `add_fractional_digits` adds. We can rework `add_fractional_digits` to not add trailing zeroes, in which case we can remove this line.